fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/SimplifyWindowExpression.java

```diff
@@ -0,0 +1,129 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.nereids.pattern.MatchingContext;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Alias;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.WindowExpression;
+import org.apache.doris.nereids.trees.expressions.functions.BoundFunction;
+import org.apache.doris.nereids.trees.expressions.literal.TinyIntLiteral;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+import org.apache.doris.nereids.trees.plans.logical.LogicalWindow;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * rewrite func(para) over (partition by unique_keys)
+ * 1. func() is count(non-null) or rank/dense_rank/row_number -> 1
+ * 2. func(para) is min/max/sum/avg/first_value/last_value -> para
+ * e.g
+ * select max(c1) over(partition by pk) from t1;
+ * -> select c1 from t1;
+ * */
+public class SimplifyWindowExpression extends OneRewriteRuleFactory {
+    private static final String COUNT = "count";
+    private static final ImmutableSet<String> REWRRITE_TO_CONST_WINDOW_FUNCTIONS =
+            ImmutableSet.of("rank", "dense_rank", "row_number");
+    private static final ImmutableSet<String> REWRRITE_TO_SLOT_WINDOW_FUNCTIONS =
+            ImmutableSet.of("min", "max", "sum", "avg", "first_value", "last_value");
+
+    @Override
+    public Rule build() {
+        return logicalWindow(any()).thenApply(this::simplify)
+                .toRule(RuleType.SIMPLIFY_WINDOW_EXPRESSION);
+    }
+
+    private Plan simplify(MatchingContext<LogicalWindow<Plan>> ctx) {
+        LogicalWindow<Plan> window = ctx.root;
+        ImmutableList.Builder<NamedExpression> projectionsBuilder = ImmutableList.builder();
+        ImmutableList.Builder<NamedExpression> remainWindowExpression = ImmutableList.builder();
+        List<NamedExpression> windowExpressions = window.getWindowExpressions();
+        for (NamedExpression expr : windowExpressions) {
+            Alias alias = (Alias) expr;
+            WindowExpression windowExpression = (WindowExpression) alias.child();
+            if (!windowExpression.getOrderKeys().isEmpty()) {
+                remainWindowExpression.add(expr);
```

Comment:
if partition key is unique, order key is not use anymore, so why remain them?

Reply:
LogicalWindow 
->
 LogicalProject(projections)
     + LogicalWindow(remainWindowExpression) 
remainWindowExpressions are the WindowExpressions that don't meet the conditions for optimization, and still need to be evaluated int LogicalWindow.

Reply:
Tthe restriction has been removed